tests/tools.vitruv.applications.pcmumlclass.tests/src/tools/vitruv/applications/pcmumlclass/tests/MediaStoreRepositoryCreationTest.xtend

```diff
@@ -165,4 +175,331 @@ class MediaStoreRepositoryCreationTest extends LegacyPcmUmlClassApplicationTest
+	 * the insertion of Uml-models and reload the view, in oder to achieve the wanted propagation of a correct ComponentRepository.
+	 * In a editor based creation process, this would automatically be done if the synchronization is called often enough (enough saves).
+	 * Because of the necessary reloads, the model is loaded (while the out-of-synch elements remain) and registered with new IDs in the UUID resolver.
+	 * This might make it necessary to provide the VM that runs the tests with additional heap space.
+	 */
+	protected def simulateRepositoryInsertion_PCM(Repository originalRepository, String pcmOutputPath,
+		String umlOutputPath) {
+
+		changePcmView[
+			userInteraction.addNextTextInput(umlOutputPath) // answers where to save the corresponding .uml model
+			registerRoot(originalRepository, Path.of(pcmOutputPath).uri)
+		]
+	}
+
+	private def simulateDataTypeInsertion_UML(String modelName, PackageableElement umlDataType) {
+		changeUmlView [
+			val repositoryPackage = UmlQueryUtil.claimUmlModel(it, modelName).nestedPackages.head
+			assertNotNull(repositoryPackage)
+			val datatypesPackage = repositoryPackage.nestedPackages.findFirst [
+				it.name == DefaultLiterals.DATATYPES_PACKAGE_NAME
+			]
+			assertNotNull(datatypesPackage)
+
+			datatypesPackage.packagedElements += umlDataType
+		]
+	}
+
+	private def simulateContractsPackageElementInsertion_UML(String modelName,
+		PackageableElement originalContractsPackageElement) {
+		changeUmlView[
+			val repositoryPackage = UmlQueryUtil.claimUmlModel(it, modelName).nestedPackages.head
+			assertNotNull(repositoryPackage)
+
+			var contractsPackage = repositoryPackage.nestedPackages.findFirst [
+				it.name == DefaultLiterals.CONTRACTS_PACKAGE_NAME
+			]
+			assertNotNull(contractsPackage)
+
+			if (!(originalContractsPackageElement instanceof Interface)) {
+				contractsPackage.packagedElements += originalContractsPackageElement
+				return
+			}
+
+			val originalInterface = originalContractsPackageElement as Interface
+			val insertedInterface = EcoreUtil.copy(originalInterface)
+
+			// add each operation without its parameters because at least the returnParameters will be already generated
+			for (operation : insertedInterface.ownedOperations) {
+				operation.ownedParameters.clear
+			}
+			contractsPackage.packagedElements += insertedInterface
+		]
+
+		if (!(originalContractsPackageElement instanceof Interface)) {
+			return
+		}
+
+		changeUmlView[
+			val repositoryPackage = UmlQueryUtil.claimUmlModel(it, modelName).nestedPackages.head
+			assertNotNull(repositoryPackage)
+			var contractsPackage = repositoryPackage.nestedPackages.findFirst [
+				it.name == DefaultLiterals.CONTRACTS_PACKAGE_NAME
+			]
+			assertNotNull(contractsPackage)
+			val originalInterface = originalContractsPackageElement as Interface
+			var generatedInterface = contractsPackage.packagedElements.filter(Interface).findFirst [
+				it.name == originalInterface.name
```

Comment:
Names of Interfaces have to be unique for this to work. We could switch to an explicit map to avoid that limitation.